js/background.js

```diff
@@ -22,6 +23,20 @@ const prepareValue = value => {
   return v.toString();
 };
 
+const updateBalance = addressJSON => {
+  storage.sync.set({balance: addressJSON.balance});
+};
+const checkWallet = wallet => {
+  if(wallet && /\d\L/.test(wallet)) getBalance(wallet);
+};
+const getBalance = wallet => {
+  storage.sync.set({wallet: wallet});
+  return fetch(walletAPIUrl + wallet)
+  .then(res => res.status === 200 && res.json())
+  .then(updateBalance);
+};
```

Comment:
It will be good to handle response with invalid wallet address and show some simple message to user when he makes a mistake with address typo